From 98fa25db65984e59d72eebcc42265bdf5e02fd95 Mon Sep 17 00:00:00 2001
From: Amos Kong <akong@redhat.com>
Date: Tue, 14 Jan 2014 09:56:39 +0100
Subject: [PATCH 15/16] fix -boot strict regressed in commit 6ef4716

RH-Author: Amos Kong <akong@redhat.com>
Message-id: <1389693400-21546-2-git-send-email-akong@redhat.com>
Patchwork-id: 56652
O-Subject: [RHEL-7.0 qemu-kvm PATCH 1/2] fix -boot strict regressed in commit 6ef4716
Bugzilla: 997817
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

Commit 6ef4716 cleaned up parsing of -boot option argument, but
accidentally dropped parameter strict.  It should have been updated
exactly like parameter menu. Do that.

Signed-off-by: Amos Kong <akong@redhat.com>
Reviewed-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>
(cherry picked from commit e5187b561ff7b241d609a2848131879072169565)
---
 vl.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 vl.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/vl.c b/vl.c
index 781c352..11899dc 100644
--- a/vl.c
+++ b/vl.c
@@ -459,7 +459,7 @@ static QemuOptsList qemu_boot_opts = {
             .type = QEMU_OPT_STRING,
         }, {
             .name = "strict",
-            .type = QEMU_OPT_STRING,
+            .type = QEMU_OPT_BOOL,
         },
         { /*End of list */ }
     },
@@ -4059,6 +4059,7 @@ int main(int argc, char **argv, char **envp)
         }
 
         boot_menu = qemu_opt_get_bool(opts, "menu", boot_menu);
+        boot_strict = qemu_opt_get_bool(opts, "strict", false);
     }
 
     if (!kernel_cmdline) {
-- 
1.7.1

